s2aff/__init__.py

```diff
@@ -115,27 +115,23 @@ def predict(self, raw_affiliations):
 
             if len(candidates) == 0:
                 output_scores_and_thresh = [self.no_candidates_output_text], [0.0]
+            elif len(candidates) == 1:
+                output_scores_and_thresh = [self.no_ror_output_text], [0.0] if scores[0] < self.pairwise_model_threshold else (candidates, scores)
+
             else:
                 reranked_candidates, reranked_scores = self.pairwise_model.predict(
                     raw_affiliation, candidates[: self.top_k_first_stage], scores[: self.top_k_first_stage]
                 )
                 # apply threshold to reranked scores
                 if len(reranked_candidates) == 0:
                     output_scores_and_thresh = [self.no_candidates_output_text], [0.0]
-                elif len(reranked_candidates) == 1:
-                    if reranked_scores[0] < self.pairwise_model_threshold:
+                elif reranked_scores[0] < self.pairwise_model_threshold and \
+                        (len(reranked_candidates) == 1 or \
```

Comment:
I don't think len(reranked_candidates) can be 1 anymore as you check right above.